### PR TITLE
feat(ingestion): move the executor engine into datahub.executor

### PR DESCRIPTION
**Pull request by rajatoss**
Moves the metadata-ingestion execution engine out of the separate `acryl-executor` distribution and into `acryl-datahub` as `datahub.executor`. `datahub-actions` — the OSS path for UI-triggered ingestion — now imports it from there, which removes its last dependency on the retired package.

The DataHub Cloud log/artifact upload does **not** move. It needs a Cloud-managed S3 bucket plus two GraphQL operations that do not exist in OSS, so it stays where it can actually run.

### Why

`acryl-executor` was a separately-versioned package that both `datahub-actions` and the DataHub Cloud remote executor depended on, while being almost entirely generic ingestion machinery: venv management, subprocess execution, and the execution request/result types. Keeping it separate meant a version-skew surface for no benefit, and it put Cloud-only code next to code that belongs in OSS.

### What changed

The port is deliberately behaviour-preserving. **22 of 32 files are logically identical** to their originals modulo the namespace rewrite, verified by AST comparison. The unavoidable changes:

- `env_utils` is split. The generic getters land in `common/env_config.py` with their bodies and environment variable names unchanged, including `ACRYL_EXECUTOR_GMS_PAYLOAD_MAX_LENGTH` — kept verbatim so deployments that already set it keep working.
- The two wrapper scripts become modules under `wrappers/` and are launched with the current interpreter by **resolved absolute path**, rather than by bare name off `PATH`. PATH lookup was fragile here, since this code also mutates `PATH`. Deliberately not `python -m`: `-m` puts the subprocess's working directory on `sys.path[0]`, so a stray module sitting there (the shipped image runs from `/tmp`) shadows a real import and kills the run before any wrapper code executes. Invoking by path puts the wrapper's own directory first instead, matching the console-script entry points this replaced.
- `ExecutionContext` gains an optional `artifact_dir` so a caller can locate a run's logs and artifacts after `execute()` returns without reconstructing the path from configuration. It is published only once the ingest subprocess exists, so a venv failure does not advertise an empty directory as collectible.
- `SubProcessIngestionTask` no longer uploads anything. It writes artifacts locally and publishes the directory; the three retired `cloud_log_bucket` / `cloud_log_path` / `cloud_log_cleanup` task-config keys are still **accepted and warned about** rather than removed, because `ConfigModel` is `extra="forbid"` — deleting them would turn a leftover key in an operator's `task_configs` into a startup failure whose message gives no hint why.
- The task registry rewrites dotted paths that still begin `acryl.executor.` and logs which path to use instead. `ExecutorConfig.task_configs` is a user-facing recipe field, so those strings exist in operator configuration we cannot migrate, and the error they would otherwise hit names a pip extra that does not exist.
- `ReportingExecutor` sets `executorInstanceId` on the result aspect when an executor declares an instance id, but **no model change is made here**: nothing in this repo sets `executor_instance_id` (it defaults to `None`, and the `executor` action does not expose it), so the key is never passed and `ExecutionRequestResult` does not declare the field. A deployment that does set it registers its own models package via the `datahub.custom_packages` entry point, which replaces these generated classes with ones that declare it. An earlier revision of this PR added the field to the aspect on the mistaken belief that the generated classes here were the ones in play; that has been reverted.

Two latent issues are **suppressed rather than fixed**, to keep the move behaviour-identical: a shared mutable default in `ExecutorContext` (`B006`) and a bare re-raise in `DefaultExecutor` (`B904`, where either fix changes traceback text that reaches users and tests). `logger.warn` → `logger.warning` is taken, since `Logger.warn` is removed in Python 3.13.

### Dependencies

Adds `loguru` and `anyio` to the base dependency set — and nothing else. **`boto3` is not needed**, so this is a net reduction versus `acryl-executor`, which required it unconditionally. `urllib3` is already in the base set on master and is not touched here.

This branch is now identical to master on `urllib3`: `>=2.5.0,<3.0` in the dependency tables, `>=2.7.0,<3.0.0` as the lock-time constraint from #19056, and `urllib3==2.7.0` resolved in `constraints.txt`. No version is downgraded and no advisory is reintroduced.

It is worth recording how it briefly diverged, because the failure mode is invisible. A merge from master into this branch resolved in favour of the branch side and dropped the lock-time constraint while keeping the `setup.py` floor, so for a few commits the branch carried a weaker guard than master. `constraints.txt` still resolved 2.7.0 throughout, but only because that is the newest release — a later `uv lock` could have fallen back toward the 2.5.0 floor. The constraint has been restored and has since survived two further merges from master.

`checkLockFile` cannot catch this: `verify_pyproject_equivalence.py` compares the dependency tables and never reads `[tool.uv] constraint-dependencies`, so the gap passed fully green CI. Worth a guard there, since the next master merge on any branch can reproduce it.

### Tests

266 cases, passing in default and random order. Four source files are dropped: one whose tests `return False` instead of asserting (so pytest reported six passes while four actually failed), two superseded by existing OSS tests, and one covering the upload that stayed behind. Twelve pre-existing tests that asserted nothing about the code under test are also dropped — called out here so it does not read as a coverage regression. Six tests were making live PyPI calls in a unit suite and are now mocked.

Added coverage for three branches nothing in the suite reached: the cancellation cleanup path in `_monitor_subprocess`, the secret masking of the structured report, and the test-connection task's `CancelledError` handler. The masking branch was structurally unreachable for its whole file, because an autouse fixture resets the secret registry around every test; each of the three was confirmed by coverage delta rather than by the test merely passing.

### Compatibility

Nothing in this PR imports `datahub.executor`, other than `datahub-actions`. The DataHub Cloud remote executor is repointed separately, once a release containing this ships.

The `acryl-datahub-actions[executor]` extra is retained (now empty) rather than removed. CI passes it via `ACTIONS_EXTRA_PACKAGES`, and the docs tell users to install it -- including the published 1.3.0 versioned docs, which cannot be corrected retroactively. pip and uv only warn on an unknown extra, so removing it would not break an install; keeping the name just avoids handing anyone a broken instruction.

### Checklist

- [x] The PR conforms to DataHub's Contributing Guideline, particularly PR Title Format
- [x] Tests for the changes have been added/updated
- [x] Docs related to the changes have been added/updated
- [x] An entry has been made in Updating DataHub

---

> [!NOTE]
> **Overview**
> **Ingestion execution** now lives in **`acryl-datahub`** under **`datahub.executor`** instead of the separate **`acryl-executor`** PyPI package. **`datahub-actions`**’ executor plugin drops the `acryl-executor` extra dependency and imports **`datahub.executor.*`**; default ingest/test-connection task types use **`datahub.executor.execution.*`** dotted paths.
> 
> The port is largely a namespace move with targeted behavior tweaks: **`ReportingExecutor`** can emit optional **`executorInstanceId`** on execution results; **`SubProcessIngestionTask`** writes artifacts locally and exposes **`ExecutionContext.artifact_dir`** instead of uploading logs (retired **`cloud_log_*`** config keys still parse with warnings); wrapper scripts run as **`datahub.executor.wrappers.*`** by **absolute path** to avoid CWD import shadowing. **`TaskRegistry`** rewrites legacy **`acryl.executor.*`** task type strings with a deprecation warning.
> 
> **Dependencies:** base **`acryl-datahub`** adds **`loguru`** and **`anyio`**; lock/constraints updated accordingly. **Docs** document retirement of **`acryl-executor`** and the optional **`executorInstanceId`** model field. **Tests** add executor integration/unit coverage (help-probe / CLI fallback, cancellation, masking) and a **`degraded`** pytest marker.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c571bf3f255a9dec097eea9cfed62fb0ec663b08. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Moves the ingestion execution engine into `datahub.executor`, retiring `acryl-executor`, so UI-triggered ingestion and connection tests run without a separate package. Preserves behavior, fixes cancellation edge cases and `uv` cache growth, and does not change OSS models.

- `datahub-actions` now imports `datahub.executor.*`; the `executor` extra remains but is empty for compatibility.
- Wrappers live under `datahub.executor.wrappers.*`, launched by resolved absolute path to avoid CWD import shadowing. The wrapper installs its SIGTERM handler before spawning the child, runs in its own process group, and on cancellation we signal the group; the wrapper then reaps its child. Venv `sys.path` isolation is preserved.
- `ExecutionContext` publishes `artifact_dir` only after a subprocess exists to avoid empty log artifacts on setup failures.
- Task registry rewrites `acryl.executor.*` task types to `datahub.executor.*` with a warning; behavior is otherwise unchanged.
- Config/deps: move env getters to `datahub.executor.common.env_config`; accept and warn on deprecated `cloud_log_bucket`, `cloud_log_path`, and `cloud_log_cleanup` keys (no effect). Add `loguru` and `anyio`; restore the `urllib3` CVE floor in constraints.
- Runner bootstrap: install named `acryl-datahub` with `--no-deps` before the real install so bundled constraints are readable and the `uv` cache does not grow unbounded per run.
- Models: no schema change in OSS. `ReportingExecutor` still sets `executorInstanceId` when models declare it; the generated OSS classes do not include this field.
- Types: annotate the executor list in `datahub-actions` to satisfy list invariance against `Executor`.

**Migration**
- No action for most users; `acryl-datahub-actions[executor]` continues to install what it needs.
- If `task_configs` reference `acryl.executor.*`, change them to `datahub.executor.*`; old paths warn and will be removed. Remove deprecated `cloud_log_*` keys.

<sup>Written for commit 5c3c64300e6e5e4a4025d4686f0194e7d9158ac0. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/datahub-project/datahub/pull/18942?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by rajatoss** [comment]
I have tried to avoid any kind of changes related to the core code of acryl-executor, it can be improved but to avoid any behaviour changes, i have kept it as it is.

File: metadata-ingestion/src/datahub/executor/execution/runner.py [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+            if is_dev_build:
+                return f"acryl-datahub{extras} @ {url}"
+            if version == VENV_VERSION_LATEST:
+                return f"acryl-datahub{extras}"
+            return f"acryl-datahub{extras}=={version}"
+
+        # Dev builds bypass the uv cache: always re-fetch a rebuilt wheel, persist nothing.
+        # This makes usage of dev packages inefficient, but prevents cache build up which causes
+        # executor pods to over-consume storage.
+        install_env = {**venv_env, "UV_NO_CACHE": "1"} if is_dev_build else venv_env
```
[the remark this answers is not in the document]
Comment (skrydal) [reply metadata-ingestion/src/datahub/executor/execution/runner.py:594]: Handled by docs, idea itself is wrong.

**Review by skrydal — APPROVED** [review]
